lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -691,8 +711,36 @@ def _on_relation_changed(self, event):
 
         self.charm.app_peer_data["mongos_hosts"] = json.dumps(self.get_mongos_hosts())
 
-    def pass_hook_checks(self, event):
+    def pass_hook_checks(self, event: EventBase):
         """Runs the pre-hooks checks for ConfigServerRequirer, returns True if all pass."""
+        proceed_complex_hook_checks = self.pass_sanity_hook_checks(event)
+        if not proceed_complex_hook_checks:
+            return False
+
+        # occasionally, broken events have no application, in these scenarios nothing should be
+        # processed.
+        if not event.relation.app and isinstance(event, RelationBrokenEvent):
+            return False
+
+        mongos_hosts = event.relation.data[event.relation.app].get(HOSTS_KEY, None)
```

Comment:
```suggestion
        mongos_hosts = event.relation.data[event.relation.app].get(HOSTS_KEY)
```
nit: .get default value defaults to `None`